system-addon/lib/TelemetryFeed.jsm

```diff
@@ -305,8 +302,10 @@ this.TelemetryFeed = class TelemetryFeed {
     );
   }
 
-  async sendEvent(eventPromise) {
-    this.telemetrySender.sendPing(await eventPromise);
+  async sendEvent(event_object) {
+    if (this._prefs.get("telemetry")) {
```

Comment:
Hm, it might not be a good idea to check this pref on every ping – I think you might want to cache this value and add a pref listener for it the way you do in `PingCentre`